### ISSUE TITLE
paginationation fails with zero rows table

### ISSUE THREAD

**Issue by shajoezhu** [insightsengineering/rtables#608]
```
basic_table() %>% split_cols_by("Species") %>% build_table(iris) %>% export_as_txt(lpp=10)
```

# TODO 
check if table has zero rows, if so, do not paginate

**Comment by shajoezhu** [comment]
Thanks Gabe, so the todo should be

check if table has zero rows, if so, return list(tt).

cc @clarkliming for awareness

**Comment by Melkiades** [comment]
I thought about this and I figure out that the only case in which this makes sense is when we have an uneven page split, i.e. when we start the split at the bottom of a page and therefore the table has only a partial (extreme is only the titles) print in the first page (at the bottom) and then the rest is in the subsequent pages. Gabe and I already discussed about "**starting the pagination at a specific line**" as a very interesting and useful feature for the future. It comes from my understanding of medical writers' needs. I think there are virtually no cases where paginating an empty table makes sense otherwise. Certainly, for the first case, we should be very cautious about where we position the footnotes as they should be ideally on every page that has specific mentions (references) and at the end for general footnotes (or on every page?). 

In other more synthetic words, I think we need to support this empty table pagination only in the vision of adding a starting line for its pagination. This would be anyway a new feature, and we need to be certain about footnotes positioning beforehand.

**Comment by clarkliming** [comment]
actually I don't care if pagination works for a empty table. What I care is, if `export_as_txt(empty_table, lpp = 100)` fails or not

**Comment by Melkiades** [comment]
> actually I don't care if pagination works for an empty table. What I care is, if `export_as_txt(empty_table, lpp = 100)` fails or not

Yep, that for sure should not fail mysteriously :D. I was more responding to Gabe about why this could happen. Sorry if it was slightly OT.

**Comment by shajoezhu** [comment]
this is taken care of by the table sanitizer?

**Comment by Melkiades** [comment]
I checked the new functionalities and here I copy the output of this fast exploration:
```r
> a <- basic_table() %>% split_cols_by("Species") %>% build_table(iris)
> a
   setosa   versicolor   virginica
——————————————————————————————————
> table_structure(a)
[ElementaryTable] root
> find_degen_struct(a)
[[1]]
[1] "root"
> validate_table_struct(a)
[1] FALSE
> assert_valid_table(a)
Error in assert_valid_table(a) : 
  Invalid table - found 1 (sub)structures which contain no data rows.
	The first occured at path:  root
  Likely Cause: Empty data or first row split on variable with only NA values
  Use sanitize_table_struct() to fix these issues
> sanitize_table_struct(a)
     setosa      versicolor      virginica  
————————————————————————————————————————————
      -- This Section Contains No Data --  
```
Now if you do add the latter to your pipeline the problem is solved:
```r
> basic_table() %>% split_cols_by("Species") %>% build_table(iris) %>% sanitize_table_struct() %>% export_as_txt(lpp=10)
[1] "     setosa      versicolor      virginica  \n————————————————————————————————————————————\n      -- This Section Contains No Data --   \n"
```
Of course if not empty works too:
```r
> basic_table() %>% split_cols_by("Species") %>% analyze("Petal.Width") %>% build_table(iris) %>% sanitize_table_struct() %>% export_as_txt(lpp=10)
[1] "       setosa   versicolor   virginica\n——————————————————————————————————————\nMean    0.25       1.33        2.03   \n"
```

**Comment by Melkiades** [comment]
Here we have two options: 
a) we sanitize automatically from `export_as_txt`
b) we rely on the user to do sanitize (and add a better error handling, i.e. with assert_valid_table)

I am inclined to do the second.
